cmd/osctl/cmd/cluster.go

```diff
@@ -140,29 +141,22 @@ func create(ctx context.Context) (err error) {
 			return err
 		}
 
-		for key := range b64Configs {
-			var configType generate.Type
+		for _, configType := range []generate.Type{generate.TypeInit, generate.TypeControlPlane, generate.TypeJoin} {
+			var generatedConfig *v1alpha1.Config
 
-			configType, err = generate.ParseType(key)
+			generatedConfig, err = generate.Config(configType, input)
 			if err != nil {
 				return err
 			}
 
-			var data string
-
-			var configStruct *v1alpha1.Config
-
-			configStruct, err = generate.Config(configType, input)
-			if err != nil {
-				return err
-			}
-
-			data, err = configStruct.String()
-			if err != nil {
-				return err
+			switch configType {
+			case generate.TypeInit:
+				configBundle.InitCfg = generatedConfig
+			case generate.TypeControlPlane:
+				configBundle.ControlPlaneCfg = generatedConfig
+			case generate.TypeJoin:
+				configBundle.JoinCfg = generatedConfig
 			}
-
-			b64Configs[key] = base64.StdEncoding.EncodeToString([]byte(data))
```

Comment:
I think we should have the v1alpha1.ConfigBundle encapsulate all this logic in `NewConfigBundle()`. That way we only need to initialize it with one line here, and if there is ever the case in the future where we may want to generate a new bundle, we don't have it all hidden away in this cobra command.